### ISSUE TITLE
Problems with animation in packed nodes

### ISSUE THREAD

**Issue by Hralok** [godotengine/godot#85982]
### Tested versions

Found at 4.2 stable mono, reproduced at 4.2.1 rc mono

### System information

Godot v4.2.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 850M (NVIDIA; 31.0.15.3203) - Intel(R) Core(TM) i7-4710HQ CPU @ 2.50GHz (8 Threads)

### Issue description

When Instantiating an animated object, I get unsatisfactory behavior. There are two cases in the video - on the left the animation was created without a RESET track, on the right with it.

https://github.com/godotengine/godot/assets/53849521/4279f5be-ca96-4bd8-a742-93aaaa7f572f

As you can see, the left option is completely broken. Right one looks like that meme...

https://github.com/godotengine/godot/assets/53849521/62e6d8f7-f335-4d37-9b58-5d44eca464d9

... but just as bad for practical purposes as the left one

### Steps to reproduce

1) Create packed scene with node2d as root and sprite2d, animation player and animation tree as children
2) Set Atlas texture to sprite2d and create animation in animation player based on changing atlas texture region. In animation tree set animation player, and as Tree root set new AnimationNodeStateMachine. Create blend tree animation and link it with start
![image](https://github.com/godotengine/godot/assets/53849521/c1bf9503-02f5-4b76-a4ec-69c4a9355bb5)
Inside bta set animation
![image](https://github.com/godotengine/godot/assets/53849521/d29b0d10-b6d2-49df-98c4-47f5a7cbc724)
3) Instantiate this prefabed/packed shit at another scene by any way - drag and drop via editor, instantiate it with code by pressing button or else

### Minimal reproduction project (MRP)

C# MRP
[MRP.zip](https://github.com/godotengine/godot/files/13625801/MRP.zip)
GD Script MRP
[MRP_GDS.zip](https://github.com/godotengine/godot/files/13627053/MRP_GDS.zip)

**Comment by TokageItLab** [comment]
I can't test it because I can't do the C# development environment, but try to turn off the Deterministic option.

**Comment by Hralok** [comment]
I spent time and prepared an mrp for gdscript
[MRP_GDS.zip](https://github.com/godotengine/godot/files/13627032/MRP_GDS.zip)

Turning off the Deterministic option doesn't help at all
InterpolationMode and UpdateMode changes made the left case (without the RESET track) work the same as the right one, however, this is still not the behavior I expect, and besides, it seems wrong to use non-discrete mode in sprite animation, so...

**Comment by TokageItLab** [comment]
It should work correctly if Interpolation mode is set correctly, Deterministic is turned off, and default values are applied to PackedScene; The initial value of TextureRegion will be Vector2(0, 0) unless Reset, so applying the default value to PackedScene and turning off Deterministic will avoid overwriting the initial value.

Also the reset animation exists to define this initial pose at load time independent of the saved state of the scene, so this is the intended behavior.

**Comment by Hralok** [comment]
Can you please make changes in MRP_GDS that you think will fix the problem and send it me back? To be honest, i didn't understand what you are talking about (English is not my first language), and I don't think that it is real solution to problem... It is not important for me, what animation should play on certain object created from packed scene. Problem is, another created object is interfering in animation of another objects, created from same packed scene. Why should animation resets on objects A1 and A2 when I creating an A3 object? Why should all of them has same animation state and acts like they are all one?

**Comment by Hralok** [comment]
So, for those who may encounter the same problem - in my case, this is indeed the expected behavior, but it is completely unrelated to Tokage's explanations. The real problem here is that I was trying to implement animation on a sprite sheet by changing the TextureRegion of AtlasTexture. 
![image](https://github.com/godotengine/godot/assets/53849521/bec136b2-e760-4f84-9dd3-98ad40c1f920)
And thus, I did not change the unique parameter of each individual instance, I changed the graphic resource that they all referred to. 
The correct solution would be to simply drag the image into the texture field and use the specialized Animation section. In order to correctly identify the frames, a very convenient system is provided here, through which animation should be implemented. 
![image](https://github.com/godotengine/godot/assets/53849521/ffc35657-4636-474d-866a-f761292b8a77)

The question is really closed.
